### DISCUSSION TITLE
Is it possible to use jupyter-ipywidgets+ipympl as a Matplotlib backend?

### DISCUSSION THREAD

**Discussion by vadimcn** [microsoft/vscode-jupyter#13716] [Questions and Answers]
Often, I need to visualize data from the debug console while debugging Python scripts on a remote server (via VSCode's SSH remote).  
While it is possible to use X11 forwarding + Matplotlib's interactive backends, this combination is often so slow as to be completely unusable.  
So I am wondering: what would it take to use jupyter-ipywidgets+ipympl as a Matplotlib backend from a regular Python script?

**Comment by DonJayamanne** [comment]
> jupyter-ipywidgets+ipympl as a Matplotlib backend from a regular Python script?

ipympl is supported today and should work,  
Please could you let me know what problems you  are running into when attempting to use that

**Reply by vadimcn** [reply]
How do I use it?   
Please note that I'd like to do ad-hoc plotting form the VSCode debug console while debugging a regular Python app, not a notebook. Basically, something similar to what's being demoed [here](https://www.youtube.com/watch?v=mZFlLkPjhXk&t=292s).

**Reply by DonJayamanne** [reply]
> How do I use it?

* Create a ntoebook
* Add some code in the cell with ipympl and matplot lib
* Then run the code and you will get the output

> lease note that I'd like to do ad-hoc plotting form the VSCode debug console w

Currently its not possible to open ipywidgets outputs outside a notebook view.
You are after this feature https://github.com/microsoft/vscode-jupyter/issues/1120
Please upvote that issue.

**Reply by vadimcn** [reply]
#1120 seems to be about moving plot from a notebook out into a separate panel?   But I would like to plot from *outside* of a notebook view, so it doesn't seem to solve my problem...
